plugins/woocommerce-admin/client/products/sections/product-shipping-section.tsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { SelectControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { Link, useFormContext } from '@woocommerce/components';
+import {
+	EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME,
+	Product,
+	ProductShippingClass,
+} from '@woocommerce/data';
+import interpolateComponents from '@automattic/interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import { getTextControlProps } from './utils';
+import { ADMIN_URL } from '../../utils/admin-settings';
+
+const DEFAULT_SHIPPING_CLASS_OPTIONS: SelectControl.Option[] = [
+	{ value: '', label: __( 'No shipping class', 'woocommerce' ) },
+	{ value: '-1', label: __( 'Standard shipping', 'woocommerce' ) },
+];
+
+function mapShippingClassToSelectOption(
+	shippingClasses: ProductShippingClass[]
+): SelectControl.Option[] {
+	return shippingClasses.map( ( { id, name } ) => ( {
+		value: `${ id }`,
+		label: name,
+	} ) );
+}
+
+export const ProductShippingSection: React.FC = () => {
+	const { getInputProps } = useFormContext< Product >();
+
+	const shippingClassOptions = useSelect( ( select ) => {
+		const { getProductShippingClasses } = select(
+			EXPERIMENTAL_PRODUCT_SHIPPING_CLASSES_STORE_NAME
+		);
+		const shippingClasses =
+			getProductShippingClasses< ProductShippingClass[] >();
```

Comment:
It appears there might be something broken with the crud data store at the moment and the above returns `null` all the time (at-least for me). This might of been something I broke :/ which I thought I had fixed.
To get around it for now you could pass an empty object query object, but I will get a fix up for this.

Reply:
Actually this might of been an issue with my build, I will get back to you on this.

Reply:
Yup, just double checked and made sure my build was correct. This is indeed working correctly as is, sorry for the confusion.